### PR TITLE
fix(#38): add sa_token_encrypted to cluster model

### PR DESCRIPTION
**Pull request by dipak-pawar**
Fixes: #38

### CODE REVIEW

**Comment by dipak-pawar** [comment]
@alexeykazakov I have addressed your comments.

File: configuration/configuration.go [hunk 1]
```diff
@@ -613,3 +619,8 @@ func (c *ConfigurationData) GetDevModePrivateKey() []byte {
 	}
 	return nil
 }
+
+// IsSaTokenEncrypted return pointer to bool
+func IsSaTokenEncrypted(encrypted bool) *bool {
```
Comment (xcoulon) [inline configuration/configuration.go]: the name of this function is confusing: you just return a pointer to the given bool, don't you?

File: migration/migration_blackbox_test.go [hunk 2]
```diff
@@ -210,3 +211,31 @@ func testMigration005AlterClusterAPIURLIndexToUnique(t *testing.T) {
...
+                     metrics_url, logging_url, app_dns, sa_token, sa_username, sa_token_encrypted, token_provider_id,
+                     auth_client_id, auth_client_secret, auth_default_scope, type)
+			VALUES ('3eb0fb9a-b7d3-479f-9ec0-e5f93c7b1e53', now(), now(), 'exhausted', 'https://token-encrypted.api.cluster.com', 'https://console.cluster.com',
+			        'https://metrics.cluster.com', 'https://login.cluster.com', 'https://app.cluster.com', 'sometoken', 'dssas-sre', false, 'pr-id',
+			        'client-id', 'cleint-scr', 'somescope', 'OSD')`)
+	require.NoError(t, err)
+
+	// check if ALL the existing rows & new rows have the default value
+	rows, err := sqlDB.Query("SELECT sa_token_encrypted FROM cluster")
+	if err != nil {
```
Comment (xcoulon) [inline migration/migration_blackbox_test.go]: `require.NoError(t, err)`

File: migration/sql-files/006-add-sa-token-encrypted-to-cluster.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+-- Store sa_token_encrypted for cluster
+ALTER TABLE cluster ADD COLUMN sa_token_encrypted boolean NOT NULL DEFAULT FALSE;
```
Comment (xcoulon) [inline migration/sql-files/006-add-sa-token-encrypted-to-cluster.sql]: please don't use `NOT NULL` in this sql statement: if we need to rollback the deployment, we'll be in troubles.

Comment (dipak-pawar) [reply migration/sql-files/006-add-sa-token-encrypted-to-cluster.sql]: I don't see reason to rollback the deployment. Do you see?

Comment (xcoulon) [reply migration/sql-files/006-add-sa-token-encrypted-to-cluster.sql]: it's just a good practice IMO

File: configuration/configuration.go (hunk 1 above)
Comment (dipak-pawar) [reply configuration/configuration.go]: ok, let me change that

**Comment by dipak-pawar** [comment]
> In the parent issue description, you mentioned that the default value for the new field/column should be false but here in this PR, it's true. Also, I'm not sure I understand how this new field/column is used at all.

@xcoulon I have updated issue description. Sorry for confusion. Let me know if it's clear to you

**Pull request merged**